src/domains/transaction/hooks/use-transaction-query-params.ts

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from "react";
+import { useLocation } from "react-router-dom";
+
+export const useTransactionQueryParams = () => {
+	const { search } = useLocation();
+
+	const originalQueryParameters = useMemo(() => new URLSearchParams(search), [search]);
+
+	const queryParameters = useMemo<Record<string, string>>(() => {
+		const result: Record<string, string> = {};
+		for (const [key, value] of originalQueryParameters.entries()) {
+			if (key !== "reset") {
+				result[key] = value;
+			}
+		}
+		return result;
+	}, [originalQueryParameters]);
+	const hasAnyParameters = useMemo(() => Object.keys(queryParameters).length > 0, [queryParameters]);
+
+	return useMemo(
+		() => ({
+			hasAnyParameters,
+			hasReset: originalQueryParameters.get("reset") === "1",
+			queryParameters,
+		}),
+		[hasAnyParameters, originalQueryParameters, queryParameters],
+	);
```

Comment:
```suggestion
	return {
		hasAnyParameters,
		hasReset: originalQueryParameters.get("reset") === "1",
		queryParameters,
	};
```
I think `useMemo` is unnecessary here because all values are already wrapped in `useMemo` (e.g. [here](https://github.com/PayvoHQ/wallet/blob/develop/src/domains/vote/hooks/use-vote-filters.ts#L144), [here](https://github.com/PayvoHQ/wallet/blob/develop/src/domains/vote/hooks/use-delegates.ts#L87),...)

Reply:
Not really, the returned object is always created new, unlike fields. 
It doesn't matter here because we don't use the return object by itself.